src/peft/tuners/loha/config.py

```diff
@@ -69,7 +72,25 @@ class LoHaConfig(LycorisConfig):
             List of modules apart from adapter layers to be set as trainable and saved in the final checkpoint.
     """
 
-    r: int = field(default=8, metadata={"help": "LoHa rank"})
+    r: int = field(default=8, metadata={"help": "LoHa rank (used for both r1 and r2 if they are not specified)"})
+    r1: Optional[int] = field(
+        default=None,
+        metadata={
+            "help": (
+                "Rank for the first Hadamard component (w1a @ w1b). "
+                "If not specified, defaults to r/2 for ABBA-style initialization, or r otherwise."
+            )
+        },
+    )
+    r2: Optional[int] = field(
+        default=None,
+        metadata={
+            "help": (
+                "Rank for the second Hadamard component (w2a @ w2b). "
+                "If not specified, defaults to r/2 for ABBA-style initialization, or r otherwise."
+            )
+        },
+    )
```

Comment:
I see that these two parameters are described in the paper, but I think it's too much to have 3 parameters to set the rank. I would suggest either:

1. Drop r1 and r2, always use r1=r2=r.
2. If it's important to have r1!=r2 as an option, I would drop r1, use r and r2 only, where r takes the role of r1 and r2 defaults to r.

Unless I'm missing something, I wouldn't use r/2 as default, as I think it's confusing that changing from LoHa to ABBA halves the effective rank.